cluster-operator/src/main/java/io/strimzi/operator/cluster/PlatformFeaturesAvailability.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster;
+
+import io.vertx.core.json.Json;
+
+import java.util.Properties;
+
+public class PlatformFeaturesAvailability {
+    private Properties properties;
+
+    static final String MAJORVERSION = "major";
+    static final String MINORVERSION = "minor";
+    static final String ISOPENSHIFT = "isOpenshift";
+    static final String NETWORKPOLICIES = "networkPolicies";
+
+    static final String KUBERNETES1_8 = "1.8";
+    static final String KUBERNETES1_9 = "1.9";
+    static final String KUBERNETES1_10 = "1.10";
+    static final String KUBERNETES1_11 = "1.11";
+
+    public PlatformFeaturesAvailability(boolean isOpenshift, String versionResponse) {
+        if (versionResponse == null) {
+            throw new IllegalArgumentException("Kubernetes version could not be determined.");
+        }
+        this.properties = new Properties();
+        Properties versionResponseProperties = Json.decodeValue(versionResponse, Properties.class);
+
+        // we need to get versions first
+        properties.put(MAJORVERSION, versionResponseProperties.get(MAJORVERSION));
+        properties.put(MINORVERSION, versionResponseProperties.get(MINORVERSION));
+        properties.put(ISOPENSHIFT, Boolean.toString(isOpenshift));
+
+        // then we can compare
+        properties.put(NETWORKPOLICIES, Boolean.toString(this.isEqualOrNewerVersionThan(KUBERNETES1_11)));
+    }
+
+    public boolean isOpenshift() {
+        return Boolean.parseBoolean(this.properties.getProperty(ISOPENSHIFT));
+    }
+
+    public int getMinorVersion() {
+        return Integer.parseInt(this.properties.getProperty(MINORVERSION));
+    }
+
+    public int getMajorVersion() {
+        return Integer.parseInt(this.properties.getProperty(MAJORVERSION));
+    }
+
+    public boolean isNetworkPolicyAvailable() {
+        return Boolean.parseBoolean(this.properties.getProperty(NETWORKPOLICIES));
+    }
+
+    public boolean isEqualOrNewerVersionThan(String version) {
+        String[] parts = version.split("\\.");
+        if (parts.length < 2) {
+            throw new IllegalArgumentException("The Kubernetes version has to have format of X.Y where X and Y are integers.");
+        }
+        return this.isEqualOrNewerVersionThan(Integer.parseInt(parts[0]), Integer.parseInt(parts[1]));
+    }
+
+    public boolean isEqualOrNewerVersionThan(int otherMajor, int otherMinor) {
```

Comment:
This is a very ugly API. Look at the call sites: It's not immediately obvious to the read that those `int` arguments mean. I think you should have a `KubernetesVersion` class, which simply represents a kube version. It should override `equals()`, `hashCode()` and implements `Comparable<KubernetesVersion>`. `KafkaVersion` does a very similar thing if you want to see a concrete example. You can have constants on this class for any versions which are needed elsewhere in the code.

Then `PlatformFeaturesAvailability` can expose a method

```
public KubernetesVersion version() { ... }
```

And callers can use something like

```
pfa.version().compareTo(KubernetesVersion.v1_11) >= 0
// or
pfa.version().equals(KubernetesVersion.v1_11)
```